docs/source/ssl.md

```diff
@@ -0,0 +1,14 @@
+# Internal SSL
+
+JupyterHub 1.0 introduces internal_ssl configuration for encryption and authentication of all internal communication.
+
+Kubespawner can mount the internal_ssl certificates as Kubernetes secrets into the jupyter user's pod.
+
+## Setup
+
+```
+c.JupyterHub.internal_ssl = True
+
+c.JupyterHub.spawner_class = 'kubespawner.KubeSpawner'
+c.KubeSpawner.internal_ssl_directory = '/srv/jupyterhub/internal-ssl/'
```

Comment:
There is `c.JupyterHub.internal_certs_location` that serves this purpose.

https://jupyterhub.readthedocs.io/en/stable/api/app.html#jupyterhub.app.JupyterHub.internal_certs_location

Ideas:
1. Trying to use c.JupyterHub.internal_certs_location instead of defining this ourselves on KubeSpawner. Perhaps we cannot reference this, but perhaps we can? I'm not sure. If we can't, I think it make sense that JupyterHub would try pass this to kubespawner.
2. If we fail, then we could perhaps try reuse this name?

I note that the default value of c.JupyterHub.internal_certs_location is "internal-ssl", and i think that assumes like you have a path to this location being decided by `c.JupyterHub.data_files_path` which defaults to a dynamic value, determined by this logic here: https://github.com/jupyterhub/jupyterhub/blob/master/jupyterhub/_data.py

Yikes, perhaps we can request backup to figure out what make sense here.

Reply:
I was referencing it via an environment variable that is made available to kubespawner, but I was explicitly asked to not tie it to JupyterHub.  Jupyter makes available the following environment variables that I was previously used instead of the explicit internal_ssl_directory.

env['JUPYTERHUB_SSL_KEYFILE']
env['JUPYTERHUB_SSL_CERTFILE']
env['JUPYTERHUB_SSL_CLIENT_CA']

My understanding is these 3 variables are always correct and can be relied on if we are willing to tie to JupyterHub, but @yuvipanda specifically asked me to not use these.